src/isEmpty/index.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Checks if an array is empty or not
+ *
+ * @param array - The array to check for
+ * @returns A boolean If the given array is empty or not
+ *
+ * @public
+ */
+export default function isEmpty<T>(val: T[]): boolean
+
+/**
+ * Checks if an object is empty or not
+ *
+ * @param object - The object to check for
+ * @returns A boolean If the given object is empty or not
+ *
+ * @public
+ */
+export default function isEmpty<T extends { [key: string]: any }>(
+  val: T
+): boolean
+
+/**
+ * @internal
+ */
+export default function isEmpty(val: any): boolean {
+  if (Array.isArray(val)) {
+    return !val.length
+  }
+  return !Object.keys(val).length
+}
```

Comment:
- I think we should also support more native structures: `string`, `Map` and `List`.

- I also see that Lodash makes an extra step and returns `true` for `false`, `null`, `undefined`, and others but I don't like this behavior. The same way behave [Ruby/Rails](https://apidock.com/rails/Object/blank%3F). Let do the same.

@ryands17 we're planning to add `js-fns/compat` submodule that will provide 100% compatibility with Lodash, so we can make API improvements in the main module. However `isEmpty` is not a case as Ruby/Rails also have this behavior. I'm saying about the compat submodule, so you can keep it in mind.

Reply:
@kossnocorp So then should I go ahead and add it for `string`, `Map`? And which one is `List`?

Reply:
@ryands17 yes, please add `string`, `Map`, ~~`List`~~ `Set` (sorry), and [everything else that Lodash considers empty](https://github.com/lodash/lodash/blob/master/test/isEmpty.js#L25-L36).

Reply:
And overloaded functions for each and `@internal` on those right?

Reply:
Yeah, I like the idea of having elaborate overloads for the ones that always return false, such as `null`, `undefined`. 👍

Reply:
Added types specified in the lodash tests. Please let me know if I have missed anything :)